src/Core/src/Core/IStartup.cs

```diff
@@ -14,4 +14,16 @@ public interface IStartup
 		/// <param name="appBuilder">Defines a class that provides the mechanisms to configure an application's dependencies.</param>
 		void Configure(IAppHostBuilder appBuilder);
 	}
+
+	/// <summary>
+	/// Allow to create a custom IAppHostBuilder instance.
+	/// </summary>
+	public interface IHostBuilderStartup
+	{
+		/// <summary>
+		/// Create and configure a builder object.
+		/// </summary>
+		/// <returns>The new instance of the IAppHostBuilder.</returns>
+		public IAppHostBuilder CreateHostBuilder();
```

Comment:
@mattleibow I've added `IHostBuilderStartup`. To create a custom HostBuilder only need to implement this interface in Startup. However, I am thinking in a better approach, use a default interface method: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/default-interface-methods

In `IStartup`, include the CreateHostBuilder method, with a base implementation.

Reply:
Does this work in classic Xamarin?

Reply:
Not, is not working for now.